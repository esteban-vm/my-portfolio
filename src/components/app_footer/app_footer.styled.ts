import Image from 'next/image'
import tw from 'tailwind-styled-components'

export const FlagImage = tw(Image)`

  inline

  w-auto

  contrast-125

  ~h-8/12

`

export const InnerWrapper = tw.div`

  inline

  select-none

  text-center

`

export const LanguageSwitch = tw.input`

  pointer-events-auto

  invisible

  relative

  cursor-pointer

  align-middle

  font-bold

  text-neon-green-darker

  outline-none

  ~w-20/24

  before:visible

  before:absolute

  before:top-1/2

  before:w-full

  before:-translate-y-1/2

  before:rounded-full

  before:border

  before:border-neon-green-light

  before:bg-transparent

  before:shadow-md

  before:shadow-neon-green-light

  before:backdrop-blur-sm

  before:content-['']

  before:~h-10/12

  after:visible

  after:absolute

  after:top-1/2

  after:box-border

  after:flex

  after:-translate-y-1/2

  after:items-center

  after:justify-center

  after:rounded-full

  after:border

  after:border-neon-green-light

  after:bg-black/80

  after:content-['ES']

  after:~size-10/12

  default:after:left-0

  checked:after:right-0

  checked:after:content-['EN']

`

export const MusicButton = tw.button`

  pointer-events-auto

  flex

  items-center

  justify-center

  rounded-md

  border

  border-neon-green-light

  bg-black/80

  text-neon-green-darker

  shadow-md

  shadow-neon-green-light

  backdrop-blur-sm

  transition-all

  ~size-10/12

  active:scale-95

  [&>svg]:size-1/2

`

export const Wrapper = tw.footer`

  container

  pointer-events-none

  absolute

  bottom-3

  left-1/2

  z-20

  flex

  -translate-x-1/2

  items-center

  justify-around

`
