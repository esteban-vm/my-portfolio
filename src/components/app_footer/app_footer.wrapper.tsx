import type { HTMLAttributes } from 'react'
import { clsx } from '@/utils'

interface WrapperProps extends HTMLAttributes<HTMLElement> {}

export default function Wrapper(props: WrapperProps) {
  const classes = clsx(
    'container pointer-events-none absolute bottom-3 left-1/2 z-20',
    'flex -translate-x-1/2 items-center justify-around text-white'
  )

  return <footer {...props} className={classes} />
}
