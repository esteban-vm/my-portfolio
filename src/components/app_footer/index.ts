export { default } from './app_footer'
