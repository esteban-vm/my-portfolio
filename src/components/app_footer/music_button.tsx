import type { ButtonHTMLAttributes } from 'react'
import { LuPlay, LuPause } from 'react-icons/lu'
import { clsx } from '@/utils'

interface MusicButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  isPlaying: boolean
}

export default function MusicButton({ isPlaying, ...rest }: MusicButtonProps) {
  const classes = clsx('pointer-events-auto flex items-center', 'justify-center rounded-full bg-orange-500 ~size-8/12')

  return (
    <button {...rest} className={classes}>
      {isPlaying ? <LuPause /> : <LuPlay />}
    </button>
  )
}
