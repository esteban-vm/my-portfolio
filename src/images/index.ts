export { default as spFlag } from '../../public/images/spain.png'
export { default as ukFlag } from '../../public/images/uk.png'
