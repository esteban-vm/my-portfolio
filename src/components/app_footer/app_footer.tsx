import { useState } from 'react'
import { LuPause, LuPlay } from 'react-icons/lu'
import { spFlag, ukFlag } from '@/images'
import * as Styled from './app_footer.styled'

export default function AppFooter() {
  const [isPlayingMusic, setIsPlayingMusic] = useState(false)

  return (
    <Styled.Wrapper>
      <Styled.MusicButton type='button' onClick={() => setIsPlayingMusic(!isPlayingMusic)}>
        <Styled.IconWrapper>
          {isPlayingMusic ? <LuPause aria-label='Pause' /> : <LuPlay aria-label='Play' />}
        </Styled.IconWrapper>
      </Styled.MusicButton>

      <Styled.InnerWrapper>
        <Styled.FlagImage alt='Spain flag' src={spFlag} />
        &nbsp;
        <Styled.LanguageSwitch type='checkbox' />
        &nbsp;
        <Styled.FlagImage alt='UK flag' src={ukFlag} />
      </Styled.InnerWrapper>
    </Styled.Wrapper>
  )
}
