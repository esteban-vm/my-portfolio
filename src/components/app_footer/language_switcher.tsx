import type { InputHTMLAttributes } from 'react'
import Image from 'next/image'
import { spFlag, ukFlag } from '@/images'
import { clsx } from '@/utils'

interface LanguageSwitcherProps extends InputHTMLAttributes<HTMLInputElement> {}

export default function LanguageSwitcher(props: LanguageSwitcherProps) {
  const classes = clsx(
    'pointer-events-auto relative cursor-pointer align-middle font-bold text-neon-green-light outline-none',
    '~w-16/24 before:absolute before:top-1/2 before:w-full before:-translate-y-1/2 before:rounded-full',
    'before:bg-neon-green-light before:content-[""] before:~h-8/12 after:absolute after:top-1/2',
    'after:box-border after:flex after:-translate-y-1/2 after:items-center after:justify-center',
    'after:rounded-full after:border-2 after:border-neon-green-light after:bg-black after:content-["ES"]',
    'after:~size-8/12 default:after:left-0 checked:after:right-0 checked:after:content-["EN"] hover:opacity-95'
  )

  return (
    <div className='inline select-none text-center'>
      <Image alt='Spain flag' src={spFlag} />
      &nbsp;
      <input {...props} className={classes} type='checkbox' />
      &nbsp;
      <Image alt='UK flag' src={ukFlag} />
    </div>
  )
}
