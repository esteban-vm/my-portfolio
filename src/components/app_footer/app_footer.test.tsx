import { render, screen } from '@/tests'
import AppFooter from './app_footer'

describe(`${AppFooter.name}:`, () => {
  it('should render correctly', () => {
    const { asFragment } = render(<AppFooter />)
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByRole('contentinfo')).toBeInTheDocument()
  })
})
