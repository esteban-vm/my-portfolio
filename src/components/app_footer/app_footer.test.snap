// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppFooter: > should render correctly 1`] = `
<DocumentFragment>
  <footer
    class="container pointer-events-none absolute bottom-3 left-1/2 z-20 flex -translate-x-1/2 items-center justify-around"
  >
    <button
      class="group pointer-events-auto rounded-full bg-neon-green-dark outline-offset-4 ~size-8/12"
      type="button"
    >
      <span
        class="flex size-full -translate-y-1 items-center justify-center rounded-full bg-black/80 transition-transform duration-200 will-change-transform group-hover:-translate-y-2 group-active:-translate-y-[2px] [&>svg]:fill-neon-green-dark [&>svg]:stroke-neon-green-dark"
      >
        <svg
          aria-label="Play"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="5 3 19 12 5 21 5 3"
          />
        </svg>
      </span>
    </button>
    <div
      class="inline select-none text-center"
    >
      <img
        alt="Spain"
        class="inline w-auto contrast-125 ~h-8/12"
        data-nimg="1"
        decoding="async"
        height="1"
        loading="lazy"
        src="/_next/image?url=%2Fpublic%2Fimages%2FSpain.png&w=16&q=75"
        srcset="/_next/image?url=%2Fpublic%2Fimages%2FSpain.png&w=16&q=75 1x"
        style="color: transparent;"
        width="1"
      />
       
      <input
        class="pointer-events-auto invisible relative cursor-pointer align-middle font-bold text-neon-green-dark outline-none ~w-16/24 before:visible before:absolute before:top-1/2 before:w-full before:-translate-y-1/2 before:rounded-full before:border before:border-neon-green-light before:bg-transparent before:shadow-md before:shadow-neon-green-light before:backdrop-blur-sm before:content-[''] before:~h-8/12 after:visible after:absolute after:top-1/2 after:box-border after:flex after:-translate-y-1/2 after:items-center after:justify-center after:rounded-full after:border after:border-neon-green-light after:bg-black/80 after:content-['ES'] after:~size-8/12 default:after:left-0 checked:after:right-0 checked:after:content-['EN']"
        type="checkbox"
      />
       
      <img
        alt="United Kingdom"
        class="inline w-auto contrast-125 ~h-8/12"
        data-nimg="1"
        decoding="async"
        height="1"
        loading="lazy"
        src="/_next/image?url=%2Fpublic%2Fimages%2FUnitedKingdom.png&w=16&q=75"
        srcset="/_next/image?url=%2Fpublic%2Fimages%2FUnitedKingdom.png&w=16&q=75 1x"
        style="color: transparent;"
        width="1"
      />
    </div>
  </footer>
</DocumentFragment>
`;
